streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -172,7 +174,14 @@ public String toString(final String indent) {
      * Flush all state stores owned by this task
      */
     void flushState() {
-        stateMgr.flush();
+        try {
+            stateMgr.flush();
+        } catch (final ProcessorStateException e) {
```

Comment:
Why do we catch it at this level? Seems we should move it into `ProcessorStateManger#flush()` ?

Reply:
Yeah, I wasn't sure... This way is a bit roundabout. On the other hand, right now, `TaskMigratedException` is only thrown by the tasks themselves or the threads that run them. It didn't seem within the responsibility of the `ProcessorStateManager` to reason about what specific exceptions imply about the state of the task.

Reply:
Fair enough. It's not ideal that we get a `ProcessorStateException` and have to call `getCause()`, but well, it is what it is.

Reply:
I very much agree. It's quite difficult to trace which exceptions would exist in different parts of the stack, and it would be nice to clean it up in the future.

Reply:
@vvcephei while working on the cleanup PR I realized that when we call `store.flush` internally we should not blindly wrap all thrown exceptions from stores, but only those not `StreamsException` -- i.e. if the error is not from the store itself (like an IOException) but from the streams library, then just throw it out directly.

Will do this small refactoring in my PR.